testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -13695,6 +13697,43 @@ private static void checkLogicalOrFunc(SqlOperatorFixture f) {
     }
   }
 
+  /**
+   * Test cases for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6111">[CALCITE-6111]
+   * Explicit cast from expression to numeric type doesn't check overflow</a>. */
+  @Test public void testOverflow() {
+    final SqlOperatorFixture f = fixture();
+    f.checkFails(String.format(Locale.US, "SELECT cast(%d+30 as tinyint)", Byte.MAX_VALUE),
+        ".* out of range", true);
```

Comment:
I tried to see what happens if I locally change expectedError to something else...
Surprisingly the tests continue passing
it seems that the reason is that this method is called https://github.com/apache/calcite/blob/0bec957071468a2e54a22519290ac101a752fcad/testkit/src/main/java/org/apache/calcite/test/SqlOperatorFixtureImpl.java#L157-L171

which means it checks for failure only if it is not runtime...
so it could be any expected message and any number under `cast` and it continues passing...

Or did I do something wrong?

Reply:
@snuyanzin, I think it should fail as you say, there seems to be something wrong.

I also tried changing `runtime` to `false` just to see, in this way it always fails (with a cast with overflow, and cast without), I guess there is something broken in the test auxiliary methods we rely on here.

The source of the problem lies outside the present PR, but it's nonetheless a blocker for merging (if confirmed), as we can't rely on the unit tests.

Maybe @mihaibudiu has more familiarity with this part of the code, and can shed some light.

Reply:
If you call the test from SqlOperatorTest it doesn't involve the evaluation.
You have to call the test from CalciteSqlOperatorTest. That class extends SqlOperatorTest but uses a test fixture which also evaluates the expressions. In the IDE you can do this easiest if you paste the following in CalciteSqlOperatorTest:

```
  @Test
  public void testOverflow() {
    super.testOverflow();
  }
```

Reply:
Thanks for the refresher Mihai, I always forget about this behaviour in some of our test classes.

I have verified the tests, they correctly fail if a non-overflowing value/expression is passed, so they do their job.

Reply:
Was able to test, thanks for clarification